app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -45,19 +46,21 @@ def default_authn_context
   end
 
   def link_identity_from_session_data
-    IdentityLinker.new(current_user, current_issuer).link_identity(ial: ial_level)
+    IdentityLinker.new(current_user, current_issuer).
+      link_identity(ial: ial_context.ial_for_identity_record)
   end
 
   def identity_needs_verification?
-    ial2_requested? && current_user.decorate.identity_not_verified?
+    ial_context.ial2_requested? && current_user.decorate.identity_not_verified?
```

Comment:
Can you rely on the delegation to the context and leave this as `ial2_requested?`?

Reply:
Yes, there's a couple places for that --- will give it a shot